components/butterfly/src/server/push.rs

```diff
@@ -82,7 +81,7 @@ fn run_loop(server: &Server, timing: &Timing) -> ! {
                 if server.member_list.pingable_mlr(&member)
                    && !server.member_list.persistent_and_confirmed_mlr(&member)
                 {
-                    let rumors = server.rumor_heat.currently_hot_rumors(&member.id);
+                    let rumors = server.rumor_keys();
```

Comment:
Yes, it sends everything all the time.  Whether it's an increase in traffic or not isn't certain, because there will (in theory) be less rumors to send.  I think testing in a large network is the only way to get answers to this.

I'll remove the member information from this though, that's a good catch.

Reply:
If we do go this route (and really something good to do regardless), I think we should try to minimize the duplicated computation going on here. As it is, a separate thread is spawned for every member, but then each thread does the identical work of serializing all the rumors before sending them. It seems like we'd save a lot of work by serializing the rumors we want to send once, then spawning threads* to send that data.

\* Spawning a limited number of workers rather than one for each member is also probably necessary for scaling very large, but that can probably come later, especially if we take this code in an async direction.

Reply:
This feels to me like a "let's measure it and see" question.  It's hard to know for sure what kind of impact this is going to have without seeing how it performs in a large network.  If it performs poorly, then certainly reworking it is appropriate.  But if the performance is acceptable or on par with what we had before, then I don't see the harm in leaving what we have.

Your point about minimizing duplicated computation is a good one though.  I'll see what can be done there.